airflow/www/static/js/dag/details/dagRun/UpstreamEvents.tsx

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useMemo } from 'react';
+import {
+  Box, Heading, Text,
+} from '@chakra-ui/react';
+
+import {
+  CodeCell, DatasetLink, Table, TaskInstanceLink, TimeCell,
+} from 'src/components/Table';
+import { useUpstreamDatasetEvents } from 'src/api';
+import type { DagRun as DagRunType } from 'src/types';
+
+interface Props {
+  runId: DagRunType['runId'];
+}
+
+const UpstreamEvents = ({ runId }: Props) => {
+  const { data: { datasetEvents }, isLoading } = useUpstreamDatasetEvents({ runId });
+
+  const columns = useMemo(
+    () => [
+      {
+        Header: 'Dataset ID',
+        accessor: 'datasetId',
+        Cell: DatasetLink,
+      },
+      {
+        Header: 'Timestamp',
+        accessor: 'timestamp',
+        Cell: TimeCell,
+      },
+      {
+        Header: 'Source Task Instance',
+        accessor: 'sourceTaskId',
+        Cell: TaskInstanceLink,
+      },
+      {
+        Header: 'Extra',
+        accessor: 'extra',
+        disableSortBy: true,
+        Cell: CodeCell,
+      },
+    ],
+    [],
+  );
+
+  const data = useMemo(
+    () => datasetEvents,
+    [datasetEvents],
+  );
```

Comment:
Ditto-isxh here. This memo seems... odd? Shouldn't the cahing from `useUpstreamDatasetEvents` mean we return the same object and so React VDOM knows there is nothing to do change?